src/inmanta/agent/agent.py

```diff
@@ -922,13 +934,32 @@ def __init__(
 
         self.agent_map: Optional[dict[str, str]] = agent_map
 
-        self.executor_manager = in_process_executor.InProcessExecutorManager(
-            environment,
-            self._client,
-            asyncio.get_event_loop(),
-            LOGGER,
-            self,
-        )
+        remote_executor = cfg.agent_executor_mode.get() == cfg.AgentExcutorMode.forking
+        can_have_remote_executor = code_loader
```

Comment:
1. code_loader = False is only used in testing
2. it allows the usage of the resourcecontainer, which are handlers that are loaded locally, by the tests. Because they are loaded locally, they can never be loaded on the remote side of the executor. Many of these tests also assume the handler to run in the same process as the test.

Reply:
I think this parameter could use a comment, at least stating that it's only for testing purposes.